### PR TITLE
feat: SchedulerJob module CURD

### PR DESCRIPTION
**Pull request by gavin1ee**
SchedulerJob module CURD

### CODE REVIEW

File: src/Contracts/MASA.Scheduler.Contracts.Server/Dtos/SchedulerJobHttpConfigDto.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace Masa.Scheduler.Contracts.Server.Dtos;
+
+public class SchedulerJobHttpConfigDto
+{
+    public HttpMethods HttpMethod { get; set; }
+
+    public string RequestUrl { get; set; } = string.Empty;
+
+    public Dictionary<string, string> HttpParameter { get; set; } = new Dictionary<string, string>();
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Dtos/SchedulerJobHttpConfigDto.cs:12]: HttpParameters

File: src/Contracts/MASA.Scheduler.Contracts.Server/Dtos/SchedulerJobHttpConfigDto.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    public Dictionary<string, string> HttpHeaders { get; set; } = new Dictionary<string, string>();
+
+    public string HttpBody { get; set; } = string.Empty;
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Dtos/SchedulerJobHttpConfigDto.cs:16]: JsonObject better?

File: src/Contracts/MASA.Scheduler.Contracts.Server/Infrastructure/Enums/HttpVerifyTypes.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Scheduler.Contracts.Server.Infrastructure.Enums;
+
+public enum HttpVerifyTypes
+{
+    StatusCode200 = 1,
+    CustomStatusCode,
+    ContentContains,
+    ContentDoesNotContain,
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Infrastructure/Enums/HttpVerifyTypes.cs:11]: ContentUnContains

File: src/Contracts/MASA.Scheduler.Contracts.Server/Requests/SchedulerJobListRequest.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    public DateTimeOffset? QueryStartTime { get; set; }
+
+    public DateTimeOffset? QueryEndTime { get; set; }
+
+    public JobTypes JobType { get; set; }
+
+    public string Origin { get; set; } = string.Empty;
+
+    public int Page { get; set; }
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Requests/SchedulerJobListRequest.cs:24]: define a PaginationRequest include Page and PageSize replace BaseRequest

File: src/Contracts/MASA.Scheduler.Contracts.Server/Validator/SchedulerJobValidator.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Scheduler.Contracts.Server.Validator;
+
+public class SchedulerJobValidator<T> : AbstractValidator<T> where T : SchedulerJobDto
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Validator/SchedulerJobValidator.cs:6]: SchedulerJobDtoValidator

Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Validator/SchedulerJobValidator.cs:6]: why AbstractValidator<T>?  has inheritance?

File: src/Services/Masa.Scheduler.Services.Server/Domain/Aggregates/Jobs/Configs/SchedulerJobAppConfig.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Scheduler.Services.Server.Domain.Aggregates.Jobs.Configs;
+
+public class SchedulerJobAppConfig
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Domain/Aggregates/Jobs/Configs/SchedulerJobAppConfig.cs:6]: lose inherit

File: src/Services/Masa.Scheduler.Services.Server/Domain/Aggregates/Jobs/Configs/SchedulerJobHttpConfig.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+namespace Masa.Scheduler.Services.Server.Domain.Aggregates.Jobs.Configs;
+
+public class SchedulerJobHttpConfig
+{
+    public HttpMethods HttpMethod { get; private set; }
+
+    public string RequestUrl { get; private set; } = string.Empty;
+
+    public Dictionary<string, string> HttpParameter { get; private set; } = new();
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Domain/Aggregates/Jobs/Configs/SchedulerJobHttpConfig.cs:12]: HttpParameters

File: src/Services/Masa.Scheduler.Services.Server/Application/Jobs/SchedulerJobQueryHandler.cs [hunk 8]
```diff
@@ -5,16 +5,57 @@ namespace Masa.Scheduler.Services.Server.Application.Jobs;
...
 
     [EventHandler]
     public async Task JobListHandleAsync(SchedulerJobQuery query)
     {
-        query.Result = await _schedulerJobRepository.GetListAsync();
+        var request = query.Request;
+
+        Expression<Func<SchedulerJob, bool>> condition = job => true;
+
+        condition = condition.And(request.IsCreatedByManual, job => job.Origin == string.Empty);
```
Comment (MayueCif) [inline src/Services/Masa.Scheduler.Services.Server/Application/Jobs/SchedulerJobQueryHandler.cs:26]: cumbersome write. following appear in pairs code。

**Review by MayueCif — CHANGES_REQUESTED** [review]
TBR

File: src/Contracts/MASA.Scheduler.Contracts.Server/Requests/PaginationRequest.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the Apache License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Scheduler.Contracts.Server.Requests;
+
+public class PaginationRequest : BaseRequest
```
Comment (MayueCif) [inline src/Contracts/MASA.Scheduler.Contracts.Server/Requests/PaginationRequest.cs:6]: delete BaseRequest in project and add  Masa.BuildingBlocks.Service.Contracts replace

**Review by MayueCif — COMMENTED** [review]
replace BaseRequest

**Review by MayueCif — APPROVED** [review]

**Comment by MayueCif** [comment]
@doddgu

File: src/Services/Masa.Scheduler.Services.Server/Services/SchedulerJobService.cs [hunk 10]
```diff
@@ -7,20 +7,58 @@ public class SchedulerJobService : ServiceBase
...
-        var jobs = await jobDomainService.QueryListAsync();
-        return Results.Ok(jobs);
+        var comman = new UpdateSchedulerJobCommand(requset);
+        await eventBus.PublishAsync(comman);
+        return Results.Ok();
+    }
+
+    public async Task<IResult> DeleteAsync(IEventBus eventBus, [FromQuery] Guid jobId)
+    {
+        var comman = new RemoveSchedulerJobCommand(jobId);
```
Comment (doddgu) [inline src/Services/Masa.Scheduler.Services.Server/Services/SchedulerJobService.cs:54]: comman?

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**